src/compiler/parser.ts

```diff
@@ -2202,7 +2207,11 @@ namespace ts {
             return token() === SyntaxKind.DotDotDotToken ||
                 isIdentifierOrPattern() ||
                 isModifierKind(token()) ||
-                token() === SyntaxKind.AtToken || token() === SyntaxKind.ThisKeyword || token() === SyntaxKind.NewKeyword ||
+                token() === SyntaxKind.AtToken || token() === SyntaxKind.ThisKeyword || isJSDocParameterStart();
+        }
+
+        function isJSDocParameterStart(): boolean {

```

Comment:
It seems weird that `isStartOfParameter` calls `isJSDocParameterStart` and not the other way around.